util/xmlutils.c

```diff
@@ -2317,3 +2317,727 @@ print_element_to_string (element_t element, GString *string)
+  hdlr->startElement = NULL;
+  hdlr->endElement = NULL;
+  hdlr->internalSubset = xml_file_iterator_internal_subset;
+  hdlr->externalSubset = xml_file_iterator_external_subset;
+  hdlr->isStandalone = xml_file_iterator_is_standalone;
+  hdlr->hasInternalSubset = xml_file_iterator_has_internal_subset;
+  hdlr->hasExternalSubset = xml_file_iterator_has_external_subset;
+  hdlr->resolveEntity = xml_file_iterator_resolve_entity;
+  hdlr->getEntity = xml_file_iterator_get_entity;
+  hdlr->getParameterEntity = xml_file_iterator_get_parameter_entity;
+  hdlr->entityDecl = xml_file_iterator_entity_decl;
+  hdlr->attributeDecl = xml_file_iterator_attribute_decl;
+  hdlr->elementDecl = xml_file_iterator_element_decl;
+  hdlr->notationDecl = xml_file_iterator_notation_decl;
+  hdlr->unparsedEntityDecl = xml_file_iterator_unparsed_entity_decl;
+  hdlr->setDocumentLocator = xml_file_iterator_set_document_locator;
+  hdlr->startDocument = xml_file_iterator_start_document;
+  hdlr->endDocument = xml_file_iterator_end_document;
+  hdlr->reference = NULL;
+  hdlr->characters = xml_file_iterator_characters;
+  hdlr->cdataBlock = xml_file_iterator_cdata_block;
+  hdlr->ignorableWhitespace = xml_file_iterator_characters;
+  hdlr->processingInstruction = xml_file_iterator_processing_instruction;
+  hdlr->comment = xml_file_iterator_comment;
+  hdlr->warning = xmlParserWarning;
+  hdlr->error = xmlParserError;
+  hdlr->fatalError = xmlParserError;
+}
+
+/**
+ * @brief Allocates a new, uninitialized XML file iterator.
+ *
+ * Free with xml_file_iterator_free.
+ *
+ * @return Opaque pointer to the XML file iterator data structure.
+ */
+xml_file_iterator_t
+xml_file_iterator_new (void)
+{
+  return g_malloc0 (sizeof (struct xml_file_iterator_struct));
+}
+
+/**
+ * @brief Initializes an XML file iterator to read from a given path.
+ *
+ * @param[in]  iterator         Pointer to the iterator to initialize.
+ * @param[in]  file_path        Path to the file to read from.
+ * @param[in]  output_depth     XML tree depth at which to return elements.
+ *
+ * @return -1 error, 0 success, 1 already initialized,
+ *         2 error opening file (errno is set to reason)
+ */
+int
+xml_file_iterator_init_from_file_path (xml_file_iterator_t iterator,
+                                       const char *file_path, int output_depth)
+{
+  if (iterator == NULL)
+    return -1;
+
+  if (iterator->initialized)
+    return 1;
+
+  memset (iterator, 0, sizeof (struct xml_file_iterator_struct));
+
+  LIBXML_TEST_VERSION
+
+  if (output_depth < 0)
+    output_depth = 0;
+  iterator->output_depth = output_depth;
+
+  iterator->file = fopen (file_path, "rb");
+  if (iterator->file == NULL)
+    return 2;
+
+  iterator->element_queue = g_queue_new ();
+
+  iterator->file_path = g_strdup (file_path);
+
+  xml_file_iterator_init_sax_handler (&(iterator->sax_handler));
+  iterator->parser_ctxt = xmlCreatePushParserCtxt (
+    &(iterator->sax_handler), iterator, NULL, 0, iterator->file_path);
```

Comment:
It could be a good idea to check for `xmlCreatePushParserCtxt` returning NULL
```suggestion
  iterator->parser_ctxt = xmlCreatePushParserCtxt (
    &(iterator->sax_handler), iterator, NULL, 0, iterator->file_path);
  if (iterator->parser_ctxt == NULL)
    return 3;
```
If so, the new error code would need to be checked in gvmd.